internal/controller/compaction/reconciler.go

```diff
@@ -59,12 +62,25 @@ const (
 	jobFailed
 )
 
+var (
+	// defaultCompactionJobCpuRequests defines the default cpu requests for the compaction job
+	defaultCompactionJobCpuRequests = resource.MustParse("600m")
+	// defaultCompactionJobMemoryRequests defines the default memory requests for the compaction job
+	defaultCompactionJobMemoryRequests = resource.MustParse("3Gi")
+	// lastJobCompletionReason is used to store the reason of the last completed job.
+	lastJobCompletionReason *string
+)
+
+// FullSnapshotTriggerFunc defines a function type for triggering a full snapshot.
+type FullSnapshotTriggerFunc func(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd) error
+
 // Reconciler reconciles compaction jobs for Etcd resources.
 type Reconciler struct {
 	client.Client
-	config      druidconfigv1alpha1.CompactionControllerConfiguration
-	imageVector imagevector.ImageVector
-	logger      logr.Logger
+	config              druidconfigv1alpha1.CompactionControllerConfiguration
+	imageVector         imagevector.ImageVector
+	logger              logr.Logger
+	FullSnapshotTrigger FullSnapshotTriggerFunc
```

Comment:
why is this required? It is a locally defined function and that should be sufficient, right?